assets/helpers/paymentIntegrations/paymentApi.js

```diff
@@ -149,18 +142,27 @@ function checkRegularStatus(participations: Participations, csrf: CsrfState): Ob
     }
   };
 
+  // Exhaustion of the maximum number of polls is considered a payment success
+  const handleExhaustedPolls = (error) => {
+    if (error === undefined) {
+      return Promise.resolve(PaymentSuccess);
```

Comment:
The existing flow distinguishes between definitive success (the backend confirmed that the payment went through) and more ambiguous cases (the Step Function is still running because one of the third-parties is down and although we expect it will work later we can't be 100% sure). 

I think we've lost that distinction here - is that intentional?

Reply:
Can you point me where that distinction is made? There is indeed a `/pending` route but it leads to the same outcome as success. On the back-end:
```
GET  /contribute/recurring/thankyou                 controllers.RegularContributions.displayFormMaybeAuthenticated()
...
GET  /contribute/recurring/pending                  controllers.RegularContributions.displayFormMaybeAuthenticated()
```
And on the frontend:
```
        <Route
          exact
          path={routes.recurringContribThankyou}
          component={() => (
            <ContributionsThankYouContainer
              marketingConsent={<MarketingConsentContainer />}
            />
          )}
        />
        <Route
          exact
          path={routes.recurringContribPending}
          component={() => (
            <ContributionsThankYouContainer
              marketingConsent={<MarketingConsentContainer />}
            />
          )}
        />
```

Reply:
Hmm, I (perhaps wrongly) presumed that since we had a different route there was a visual distinction too. I'm not sure if making this distinction is helpful for tracking (or any other purpose) @rupertbates @Ap0c?

Reply:
Yeah, there is some tracking done at the polling site, that's the only difference I could notice.

Reply:
I guess we would also be able to see the number of pageview hits to `/pending` and to `/thankyou`

Reply:
doesn't the user see a different message when the payment success is not definitive? like "look out for an email confirming your payment"? I'm sure I've seen that somewhere...

Reply:
@joelochlann You get that message regardless

Reply:
@jranks123 has wisely pointed out that we might get back genuine errors before the timeout elapses, which we definitely want the user to know about. so ignore me